### PR TITLE
Suppress expected errors in navigation-related unit tests

### LINKED ISSUE #80616: Error spam when running the test suite
### Godot version

4.2.dev (

### System information

Mageia 9 - Vulkan (Forward+) - dedicated AMD Radeon RX Vega M GL Graphics (RADV VEGAM) () - Intel(R) Core(TM) i7-8705G CPU @ 3.10GHz (8 Threads)

### Issue description

When compiling Godot with `dev_build=yes tests=yes`, running the test suite succeeds, but prints a lot of errors:
```
$ ./bin/godot.linuxbsd.editor.dev.x86_64 --test
[doctest] doctest version is "2.4.11"
[doctest] run with "--help" for options
ERROR: Condition "singleton != nullptr" is true.
   at: NavigationServer3D (./servers/navigation_server_3d.cpp:189)
ERROR: Resource was not pre cached for the resource section, bug?
   at: _write_resource (./scene/resources/resource_format_text.cpp:1848)
WARNING: Couldn't load resource (no cache): local://Resource_q0hwh
     at: parse_variant (./core/io/resource_format_binary.cpp:411)
ERROR: The object does not have any 'meta' values with the key 'next'.
   at: get_meta (./core/object/object.cpp:926)
ERROR: The object does not have any 'meta' values with the key 'next'.
   at: get_meta (./core/object/object.cpp:926)
ERROR: Condition "map == nullptr" is true. Returning: Vector3()
   at: map_get_closest_point (./modules/navigation/godot_navigation_server.cpp:241)
ERROR: Condition "map == nullptr" is true. Returning: Vector3()
   at: map_get_closest_point_normal (./modules/navigation/godot_navigation_server.cpp:248)
ERROR: Condition "map == nullptr" is true. Returning: RID()
   at: map_get_closest_point_owner (./modules/navigation/godot_navigation_server.cpp:255)
ERROR: Condition "map == nullptr" is true. Returning: Vector3()
   at: map_get_closest_point_to_segment (./modules/navigation/godot_navigation_server.cpp:234)
ERROR: Condition "map == nullptr" is true. Returning: Vector3()
   at: map_get_closest_point_to_segment (./modules/navigation/godot_navigation_server.cpp:234)
ERROR: Condition "map == nullptr" is true. Returning: Vector<Vector3>()
   at: map_get_path (./modules/navigation/godot_navigation_server.cpp:227)
ERROR: Condition "map == nullptr" is true. Returning: Vector<Vector3>()
   at: map_get_path (./modules/navigation/godot_navigation_server.cpp:227)
ERROR: Condition "map == nullptr" is true. Returning: r_query_result
   at: _query_path (./modules/navigation/godot_navigation_server.cpp:1161)
ERROR: Condition "!map" is true. Returning: Vector3()
   at: get_connection_pathway_end (./modules/navigation/nav_region.cpp:98)
ERROR: Condition "!map" is true. Returning: Vector3()
   at: get_connection_pathway_start (./modules/navigation/nav_region.cpp:92)
WARNING: NavigationServer3D::region_bake_navigation_mesh() is deprecated due to core threading changes. To upgrade existing code, first create a NavigationMeshSourceGeometryData3D resource. Use this resource with method parse_source_geometry_data() to parse the SceneTree for nodes that should contribute to the navigation mesh baking. The SceneTree parsing needs to happen on the main thread. After the parsing is finished use the resource with method bake_from_source_geometry_data() to bake a navigation mesh..
     at: region_bake_navigation_mesh (./modules/navigation/godot_navigation_server.cpp:468)
===============================================================================
[doctest] test cases:     816 |     816 passed | 0 failed | 1 skipped
[doctest] assertions: 2331757 | 2331757 passed | 0 failed |
[doctest] Status: SUCCESS!
WARNING: ObjectDB instances leaked at exit (run with --verbose for details).
     at: cleanup (./core/object/object.cpp:2049)
```

Some of those errors are intentionally caused by the tests, but they need to be silenced with `ERR_PRINT_OFF` and `ERR_PRINT_ON` macros in the test.

CC @smix8 as most are navigation related.

### Steps to reproduce

- `scons dev_build=yes tests=yes`
- `godot --test`

### Minimal reproduction project

n/a

### PR DESCRIPTION
**Pull request by Scony**
as in title - fixes partially #80616 i.e. it fixes the navi part

### CODE REVIEW

File: tests/test_main.cpp [hunk 1]
```diff
@@ -242,7 +242,9 @@ struct GodotTestCaseListener : public doctest::IReporter {
 			physics_server_2d = PhysicsServer2DManager::get_singleton()->new_default_server();
 			physics_server_2d->init();
 
+			ERR_PRINT_OFF
 			navigation_server_3d = NavigationServer3DManager::new_default_server();
+			ERR_PRINT_ON
```
Comment (smix8) [inline tests/test_main.cpp:247]: Can you move this down one line and include the 2D server as well, else I need to change that line again when the 2D navmesh baking pr adds the NavigationServer2DDummy.

File: tests/test_main.cpp [hunk 2]
```diff
@@ -269,7 +271,9 @@ struct GodotTestCaseListener : public doctest::IReporter {
 		}
 
 		if (suite_name.find("[Navigation]") != -1 && navigation_server_2d == nullptr && navigation_server_3d == nullptr) {
+			ERR_PRINT_OFF
 			navigation_server_3d = NavigationServer3DManager::new_default_server();
+			ERR_PRINT_ON
```
Comment (smix8) [inline tests/test_main.cpp:276]: same as above

**Review by smix8 — APPROVED** [review]

File: tests/servers/test_navigation_server_3d.h [hunk 3]
```diff
@@ -123,6 +123,7 @@ TEST_SUITE("[Navigation]") {
 		CHECK_FALSE(map.is_valid());
 
 		SUBCASE("Queries against invalid map should return empty or invalid values") {
+			ERR_PRINT_OFF
```
Comment (akien-mga) [inline tests/servers/test_navigation_server_3d.h:126]: It's not technically needed, but style wise we prefer adding semicolons after such macros.

**Review by YuriSizov — APPROVED** [review]

**Comment by YuriSizov** [comment]
Thanks!

**Pull request merged**